lec10-modelling.Rmd

```diff
@@ -1,4 +1,637 @@
+
+> ### Lesson outline
+> 
+> Total lesson time: 2 hours
+>
+> - Review eigenvalues and eigenvectors (10 minutes)
+> - Vectorizing linear systems of differential equations: another perspective on eigenvalues (20 minutes)
+> - Bifurcations (20 minutes)
+> - Fitting models to data 
+>     - Fitting simulated data for a single parameters (20 minutes)
+>     - Fitting real data for two parameters (40 minutes)
+
+> ### Setup
+> 
+> - Download [this plant biomass dataset](https://github.com/UofTCoders/council/blob/master/datasets/joel/plant-biomass-preprocess.csv) and put it in your working directory.
+> - `install.packages('lattice')`
+> - `install.packages('phaseR')`
+> - `install.packages('deSolve')`
+> - `install.packages('ggplot2')` (or `tidyverse`)
+> - `install.packages('dplyr')` (or `tidyverse`)
+> - `install.packages('tidyr')` (or `tidyverse`)
+
+## Recap: quantitative analysis of fixed points and stability
+
+Here are the steps to quantitatively analyze a model. These steps are written for a
+two-dimensional system, but they can be applied to a system of any number of dimensions.
+
+This is a generic two-dimensional system. There are two differential equations, and 
+each is a function of both $x$ and $y$. 
+
+$$ \frac{dx}{dt} = f(x, y) \\
+\frac{dy}{dt} = g(x,y)$$
+
+1. **Fixed points**: Find all combinations of $x$ and $y$ that satisfy $f(x, y) = 0$
+and $g(x, y) = 0$.
+
+2. **Stability**: Calculate the Jacobian matrix and find its eigenvalues for each
+fixed point. This is the Jacobian matrix:
+
+$$J=\begin{bmatrix}
+    \frac{\partial f}{\partial x} & \frac{\partial f}{\partial y} \\
+    \frac{\partial g}{\partial x} & \frac{\partial g}{\partial y}
+\end{bmatrix}$$
+
+- find eigenvalues at each fixed point using the function `eigen` in R, or by 
+solving the equation $\text{Det}(A - \lambda \mathbb{1}) = 0$).
+- for a set of eigenvalues ($\lambda_1$, $\lambda_2$):
+    - If all eigenvalues have negative real parts, the fixed point is **stable**.
+    - If one or more eigenvalues has a positive real part, the fixed point is **unstable**.
+    - If the eigenvalues have an imaginary (**complex**) part, the fixed point is **oscillatory**.
+
+## Vectorizing systems of differential equations
+
+To see why eigenvalues tell us about the stability of a fixed point, let's look
+at a linear system of diferential equations and its solution.
+
+Consider this system of differential equations:
+
+$$\frac{d x_1}{dt}=Ax_1+Bx_2 \\
+\frac{d x_2}{dt} = Cx_1+Dx_2$$
+
+This system is **linear**: each term only has either $x_1$ or $x_2$ in it, and there
+are no terms that have higher powers of $x_1$ and $x_2$ (such as $x_1 x_2$ or $x_1^2$).
+
+Because it's linear, this system can be written as a matrix equation:
+
+$$\frac{d \vec{x}}{dt} = A\vec{x}$$  
+
+where $\vec{x}$ is two-element vector $(x_1,x_2)$, $\frac{d \vec{x}}{dt}$ is also a
+two-element vector $(\frac{d x_1}{dt}, \frac{d x_2}{dt})$, and 
+
+$$A=\begin{bmatrix}
```

Comment:
Maybe denote the matrix with `E` instead of `A`, might be confusing that `A` is already used in `Ax1`?